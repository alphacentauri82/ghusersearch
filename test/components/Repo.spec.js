import React from 'react';
import renderer from 'react-test-renderer';
import MockDate from 'mockdate';
import {
  StyleSheetTestUtils,
} from 'aphrodite';

import Repo from 'components/Repo';

describe('Component: Repo', () => {

  beforeAll(() => {
    MockDate.set(new Date('2017-04-04T09:00:00Z'));
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    MockDate.reset();
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render all data items', () => {
    const component = renderer.create(
      <Repo
        description="A description"
        fork={true}
        html_url="https://github.com"
        language="JS"
        name="some repo"
        stargazers_count={10}
        pushed_at="2017-04-01T09:00:00Z"
        forks_count={2}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should not render anything for optional data items', () => {
    const component = renderer.create(
      <Repo
        fork={false}
        html_url="https://github.com"
        name="some repo"
        stargazers_count={10}
        pushed_at="2017-04-01T09:00:00Z"
        forks_count={2}
      />
    );
    expect(component).toMatchSnapshot();
  });

});
