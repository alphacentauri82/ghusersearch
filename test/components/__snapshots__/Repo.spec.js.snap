// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Repo should not render anything for optional data items 1`] = `
<div>
  <p
    className="Repo_name_y4kf6o u-flex u-flexAlignItemsBaseline"
  >
    <a
      href="https://github.com"
    >
      some repo
    </a>
  </p>
  <div
    className="Repo_footer_de3o52 u-flex u-flexWrap u-flexAlignItemsCenter"
  >
    <div
      className="Repo_footerItem_vaj62s u-flex u-flexAlignItemsCenter"
    >
      <svg />
      <p>
        10
      </p>
    </div>
    <div
      className="Repo_footerItem_vaj62s u-flex u-flexAlignItemsCenter"
    >
      <svg />
      <p>
        2
      </p>
    </div>
    <p
      className="u-sm-flexExpandLeft"
    >
      last push 
      3 days
       ago
    </p>
  </div>
</div>
`;

exports[`Component: Repo should render all data items 1`] = `
<div>
  <p
    className="Repo_name_y4kf6o u-flex u-flexAlignItemsBaseline"
  >
    <a
      href="https://github.com"
    >
      some repo
    </a>
    <span
      className="Repo_fork_1im0xn8"
    >
      (fork)
    </span>
  </p>
  <p
    className="Repo_desc_ylvipy"
  >
    A description
  </p>
  <div
    className="Repo_footer_de3o52 u-flex u-flexWrap u-flexAlignItemsCenter"
  >
    <p
      className="Repo_footerItem_vaj62s"
    >
      JS
    </p>
    <div
      className="Repo_footerItem_vaj62s u-flex u-flexAlignItemsCenter"
    >
      <svg />
      <p>
        10
      </p>
    </div>
    <div
      className="Repo_footerItem_vaj62s u-flex u-flexAlignItemsCenter"
    >
      <svg />
      <p>
        2
      </p>
    </div>
    <p
      className="u-sm-flexExpandLeft"
    >
      last push 
      3 days
       ago
    </p>
  </div>
</div>
`;
